CODING_GUIDELINES.rst

```diff
@@ -0,0 +1,89 @@
+==================
+Codeing Guidelines
+==================
+
+Code design
+-----------
+Before accepting changes we assess the design according to the following guidelines
+
+0. Look at each piece of code and ask yourself: Would I point to this in a job
+   interview as an example of well crafted code according to my best abilities?
+
+1. If it isn't tested and verified, it isn't scientific. This is a scientific
+   software library.
+
+2. Separate concerns: any unit of code should have one concern and one only.
+   The implementation details of any unit of code should not be known to any
+   other unit of code.
+
+3. Do not repeat yourself (DRY):
+   `https://en.wikipedia.org/wiki/Don%27t_repeat_yourself`_. Always check if
+   functionality exists elsewhere before implementing. Also recognize when
+   multiple specific implementations of an algorithm can be replaced with a
+   single generic one.
+
+4. Do not reinvent the wheel: Instead of writing a framework to solve a problem
+   consider existing solutions.
+
+5. Names should state the intent of code objects. If the intent of a code
+   expression **might** not be immediately obvious it deserves a name which
+   makes it obvious.
+
+Docstrings
+----------
+In general we consider that a well named simple function with type annotations
+does not require a docstring.  When a long-form docstring is appropriate,
+use `NumPy format <https://developer.lsst.io/python/numpydoc.html>`__. Prefer
+python type annotations over describing parameter / return types in the
+docstring.
+
+- Consider new public classes / functions: they should have a docstring if
```

Comment:
These are bulleted while the items in "Code design" are enumerated. Is there a reason for that?